source/deployment-guide/mobile/configure-microsoft-intune-mam.rst

```diff
@@ -0,0 +1,398 @@
+
+* Create an iOS App Protection Policy.
+* Add the Mattermost bundle ID.
+* Assign the policy using Azure AD groups.
+
+Step 5: Validate Using the Mobile App
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Ensure test users are assigned in Intune and properly licensed.
+* Test enrollment from an iOS device.
+* Confirm enforcement behaviors.
+* Verify mid-session enforcement behavior.
+
+Identity Configuration for Intune MAM
+--------------------------------------
+
+This section defines the identity requirements, constraints, and runtime behavior for using Microsoft Intune MAM with Mattermost.
+
+All identity prerequisites must be met before enabling Intune MAM or enrolling users.
+
+Required Identity Model
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Microsoft Intune MAM for Mattermost requires Azure AD ``objectId`` as the authoritative user identifier.
+
+* No alternative identifiers are supported.
+* If identity is misconfigured, Intune MAM enrollment will fail, even if all other configuration steps are correct.
+* There is no fallback or partial enforcement mode.
+* This requirement applies regardless of authentication method.
+
+Identity Consistency Requirements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The same Azure AD ``objectId`` must resolve consistently across:
+
+* Mobile authentication (OAuth / MSAL)
+* Web authentication (SAML)
+* Directory sync (LDAP / Entra ID Domain Services)
+
+``IdAttribute`` is the Mattermost Server configuration that specifies which user attribute contains the Azure AD ``objectId``.
+
+The following rules apply:
+
+* ``IdAttribute`` must equal Azure AD ``objectId``.
+* MSAL access tokens must include the ``oid`` claim.
+* All authentication systems must resolve to the same value.
+
+If any authentication path resolves a different identifier, enrollment will fail.
+
+Supported Identity Attributes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Only the identity attributes listed below are supported for Intune MAM.
+
++-------------------+------------------+------------------------------+
+| Attribute         | Supported        | Result                       |
++===================+==================+==============================+
+| objectId          | Required         | Works                        |
++-------------------+------------------+------------------------------+
+| email             | Not supported    | Enrollment fails             |
++-------------------+------------------+------------------------------+
+| preferred_username| Not supported    | Identity mismatch            |
++-------------------+------------------+------------------------------+
+| objectGUID        | Not supported    | Breaks mobile authentication |
++-------------------+------------------+------------------------------+
+| Custom attributes | Not supported    | Unsupported by Intune        |
++-------------------+------------------+------------------------------+
+
+Attribute Synchronization & Access Enforcement
```

Comment:
not sure if it should be mentioned here or somewhere else.

If Auth Provider is `SAML` users that have not been provisioned must do so first using the web or desktop app, failed to do this will result in an error message on the mobile app asking users to first login to their account using web or desktop.

Let me know if this point is not clear.

Reply:
What is the exact user-facing error string shown in the mobile app when this happens?